app/store/store.js

```diff
@@ -46,180 +43,132 @@ const setTransforms = [
+        } else if (!action.meta.offline.commit) {
+            throw new Error('Offline Action: commit action must be present.');
+        }
+
+        return effect();
+    },
+    persist: (store, options) => {
+        const persistor = persistStore(store, {storage: AsyncStorage, ...options}, () => {
+            store.dispatch({
+                type: General.STORE_REHYDRATION_COMPLETE,
             });
+        });
 
-            let purging = false;
-
-            // check to see if the logout request was successful
-            store.subscribe(async () => {
-                const state = store.getState();
-                const config = getConfig(state);
+        store.subscribe(async () => {
+            const state = store.getState();
+            const config = getConfig(state);
 
-                if (getSiteUrl() !== config?.SiteURL) {
-                    setSiteUrl(config.SiteURL);
-                }
-
-                if (state.views.root.purge && !purging) {
-                    purging = true;
-
-                    await persistor.purge();
-
-                    const resetState = getStateForReset(initialState, state);
-
-                    store.dispatch(batchActions([
-                        {
-                            type: General.OFFLINE_STORE_RESET,
-                            data: resetState,
-                        },
-                        {
-                            type: ErrorTypes.RESTORE_ERRORS,
-                            data: [...state.errors],
-                        },
-                        {
-                            type: GeneralTypes.RECEIVED_APP_DEVICE_TOKEN,
-                            data: state.entities.general.deviceToken,
-                        },
-                        {
-                            type: GeneralTypes.RECEIVED_APP_CREDENTIALS,
-                            data: {
-                                url: state.entities.general.credentials.url,
-                            },
-                        },
-                        {
-                            type: ViewTypes.SERVER_URL_CHANGED,
-                            serverUrl: state.entities.general.credentials.url || state.views.selectServer.serverUrl,
-                        },
-                        {
-                            type: GeneralTypes.RECEIVED_SERVER_VERSION,
-                            data: state.entities.general.serverVersion,
-                        },
-                        {
-                            type: General.STORE_REHYDRATION_COMPLETE,
-                        },
-                    ], 'BATCH_FOR_RESTART'));
-
-                    setTimeout(() => {
-                        purging = false;
-                        EventEmitter.emit(NavigationTypes.RESTART_APP);
-                    }, 500);
-                }
-            });
+            if (getSiteUrl() !== config?.SiteURL) {
+                setSiteUrl(config.SiteURL);
+            }
+        });
```

Comment:
why are we keeping this?

Reply:
Simply because I don't know what problem it's trying to solve. Yet.

This subscription is unrelated to app cache purge. I figured best to deal with as a separate, future item. Likely, just a quick removal... but didn't want to add this as an additional testing variable now.